packages/mux-player/src/index.ts

```diff
@@ -1145,6 +1145,19 @@ class MuxPlayerElement extends VideoApiElement {
     this.media.metadata = { ...this.metadataFromAttrs, ...val };
   }
 
+  async addCuePoints<T = any>(cuePoints: { timestamp: number; value: T }[]) {
+    if (!this.media) return Promise.reject(new Error('Cannot add cuePoints until mux-player has initialized'));
```

Comment:
comment(non-blocking): would be nice if we had a ready event or something so that this could be handled automatically for users 🙂

Reply:
I'm not sure this should be blocking. This is really an edge case that we're accounting for. A few callouts here:

> a simple promise that resolves once this.media is available

1. I'm not sure this is "a simple promise", unless I'm missing something. What are we monitoring for when the promise should resolve? `#init()` doesn't have a direct event to listen for. We could add a poll, or we listen for an indirect event, or a few other paths, but I'm not seeing a 1:1 for this case. Moreover, in the current basic implementation, `addCuePoints()` is tied to a `src`/`playback-id` (See docs, examples, and tests), which brings up:
2. Even if we refactored this to solve this case, I'm not sure the proposal would be sufficient and may confuse the user, since adding CuePoints before a src has been loaded will get "blown away" once the src has loaded (or more specifically, once `playback-core` has been `initialize()`ed. While there is a path forward for solving these cases as well, that's feeling like scope creep for this first pass addition of CuePoint support based on prior conversations.

Reply:
true, a promise does add more complex logic. 
(something like this https://github.com/luwes/super-media-element#usage)

calling `this.#init()` in `addCuePoints()` would be even simpler to make sure media is defined.

2. true

Reply:
awesome I like the `this.#init()` idea. I'll be sure to add it.